File: AVCaptureDemo/HLCapture/HLCaptureView.m
//
//  HLCaptureView.m
//  AVCaptureDemo
//
//  Created by yunfu on 2019/2/28.
//  Copyright © 2019 yunfu. All rights reserved.
//

#import "HLCaptureView.h"

@implementation HLCaptureView


- (void)setSession:(AVCaptureSession *)session{
    [(AVCaptureVideoPreviewLayer *)self.layer setSession:session];
    _session = session;
}


+ (Class)layerClass{
    return [AVCaptureVideoPreviewLayer class];
}

@end

File: AVCaptureDemo_Swift/HLCapture/HLCaptureView.swift
//
//  HLCaptureView.swift
//  AVCaptureDemo_Swift
//
//  Created by yunfu on 2019/3/1.
//  Copyright © 2019 yunfu. All rights reserved.
//

import UIKit
import AVFoundation

class HLCaptureView: UIView {

    override open class var layerClass: AnyClass {
        return AVCaptureVideoPreviewLayer.self
    }
    
    public var session:AVCaptureSession?{
        didSet{
            (self.layer as! AVCaptureVideoPreviewLayer).session = session
        }
    }
    

}
